File: tools/sms/src/main/java/top/bettercode/summer/tools/sms/b2m/B2mResponse.java
package top.bettercode.summer.tools.sms.b2m;

import java.util.List;
import top.bettercode.summer.tools.lang.property.PropertiesSource;
import top.bettercode.summer.tools.sms.SmsResponse;

/**
 * @author Peter Wu
 */
public class B2mResponse<T> implements SmsResponse {

  public static final String SUCCESS = "SUCCESS";
  private static final PropertiesSource codeMessageSource = PropertiesSource.of("b2m-message");
  private String code;

  private List<T> data;

  public B2mResponse() {
  }

  public B2mResponse(List<T> data) {
    this.code = SUCCESS;
    this.data = data;
  }

  //--------------------------------------------

  public String getCode() {
    return code;
  }

  public void setCode(String code) {
    this.code = code;
  }

  public List<T> getData() {
    return data;
  }

  public void setData(List<T> data) {
    this.data = data;
  }

  //--------------------------------------------

  @Override
  public boolean isOk() {
    return SUCCESS.equals(code);
  }

  @Override
  public String getMessage() {
    return codeMessageSource.getOrDefault(code, code);
  }

  //--------------------------------------------

  public static String getMessage(String code) {
    return codeMessageSource.getOrDefault(code, code);
  }
}

File: tools/sms/src/main/kotlin/top/bettercode/summer/tools/sms/b2m/B2mResponse.kt
package top.bettercode.summer.tools.sms.b2m

import top.bettercode.summer.tools.lang.property.PropertiesSource.Companion.of
import top.bettercode.summer.tools.sms.SmsResponse

/**
 * @author Peter Wu
 */
class B2mResponse<T> : SmsResponse {
    //--------------------------------------------
    var code: String? = null
    var data: List<T>? = null

    constructor()
    constructor(data: List<T>?) {
        this.code = SUCCESS
        this.data = data
    }

    override val isOk: Boolean
        //--------------------------------------------
        get() = SUCCESS == code
    override val message: String
        get() = codeMessageSource.getOrDefault(code!!, code!!)

    companion object {
        const val SUCCESS = "SUCCESS"
        private val codeMessageSource = of("b2m-message")

        //--------------------------------------------
        fun getMessage(code: String?): String? {
            return code?.let { codeMessageSource.getOrDefault(code, it) }
        }
    }
}
